lib/core/src/animation/easing.rs

```diff
@@ -0,0 +1,122 @@
+//! This module implements easing formulas and related utilities. To learn more about easing
+//! functions, please refer to the following link: https://easings.net/en
+
+// TODO: It would be useful to make the easing functions polymorphic. Maybe we can make use of
+// nalgebra::RealField?
+
+use core::f32::consts::PI;
+
+/// Easing function signature.
+pub trait FnEasing = Fn(f32) -> f32 + 'static;
```

Comment:
I don't like this style of writing this code. It looks like the `'static` was attached to `f32`. Write `pub trait FnEasing = (Fn(f32) -> f32) + 'static;` or `pub trait FnEasing = 'static + Fn(f32) -> f32;` instead :)